blocks/identity-block/components/social-sign-on/utils/useSocialSignIn.js

```diff
@@ -54,39 +83,96 @@ function useSocialSignIn(redirectURL, isOIDC, socialSignOnIn, onError = () => {}
 							window.location = validatedURL;
 						}
 					}),
-				auto_select: true,
+				auto_select: false,
+				state_cookie_domain: "none",
 			};
+			// one-tap Google Sign-In
+			window.google.accounts.id.initialize(googleIdConfigOneTap);
 
-			window.google.accounts.id.initialize(googleIdConfig);
-			const googleTextType = socialSignOnIn === SIGN_UP ? 'signup_with' : 'signin_with';
-			window.google.accounts.id.renderButton(document.getElementById("google-sign-in-button"), {
-				type: "standard",
-				theme: "outline",
-				size: "large",
-				text: googleTextType,
-				shape: "rectangular",
-				logo_alignment: "left",
-				width: "400",
-			});
+			if (!customButtons) {
+				const googleTextType = socialSignOnIn === SIGN_UP ? "signup_with" : "signin_with";
+				window.google.accounts.id.renderButton(document.getElementById("google-sign-in-button"), {
+					// render a standard Google Sign-In button
+					type: "standard",
+					theme: "outline",
+					size: "large",
+					text: googleTextType,
+					shape: "rectangular",
+					logo_alignment: "left",
+					width: "400",
+				});
+			}
 
 			Identity.isLoggedIn().then((isLoggedIn) => {
 				if (!isLoggedIn) {
+					// If user rejects it will not appear again based on exponential cooldown https://developers.google.com/identity/gsi/web/guides/features#exponential_cooldown
 					window.google.accounts.id.prompt();
 				} else if (isLoggedIn && isOIDC) {
 					loginByOIDC();
 				}
 			});
 		}
-	}, [config.googleClientId, Identity, isGoogleLoaded, isOIDC, loginByOIDC, redirectURL, socialSignOnIn]);
+	}, [
+		config.googleClientId,
+		Identity,
+		isGoogleLoaded,
+		isOIDC,
+		loginByOIDC,
+		redirectURL,
+		socialSignOnIn,
+		customButtons,
+	]);
 
 	useEffect(() => {
 		const initializeFacebook = async () => {
-			await Identity.initFacebookLogin(null);
+			// Basic setup: https://developers.facebook.com/docs/javascript/quickstart/
+			await Identity.initFacebookLogin(null, null, customButtons);
 		};
 		if (config.facebookAppId) {
 			initializeFacebook();
 		}
-	}, [Identity, config.facebookAppId]);
+	}, [Identity, config.facebookAppId, customButtons]);
+
+	useEffect(() => {
+		const checkIsLoggedIn = () => {
+			setTimeout(() => {
+				Identity.isLoggedIn().then((isLoggedIn) => {
+					if (isLoggedIn) {
+						const validatedURL = validateURL(redirectURL);
+						window.location = validatedURL;
+					}
+				});
+			}, 1500);
+		};
+
+		const setIdentityUpdated = () => {
+			setTimeout(() => {
+				setUpdateIdentities((prev) => !prev);
+			}, 1500);
+		};
+
+		if (customButtons && isFBInitialized && !isFBEventSubscribed && window?.FB?.Event) {
+			setIsFBEventSubscribed(true);
+			window.FB.Event.subscribe("auth.statusChange", (response) => {
+				if (response.status === "connected") {
+					if (fromSocialSignOnBlock) {
+						checkIsLoggedIn();
+					} else {
+						setIdentityUpdated();
+					}
+				}
+			});
+		}
```

Comment:
This one's complex... you have the timer from above, but you also have this conditional timer that should have a cleanup in the same useEffect.  Is this useEffect doing too much?  Would this be better served as two effects?

Reply:
You could clear both timers (it doesn't hurt to clear an undefined timer). You could also get rid of the function (it looks like it's only being called here) and since it's an event that requires the status change, do we even need the timeout?  Can't you just setUpdateIdentities() to some value here (line 161)?  Why does it have to wait a second and a half before changing it?

Reply:
it doesn't hurt to have multiple effects (it can actually be super helpful to make them laser focused because it doesn't run all the other code every time one or the other changes)

Reply:
Of course, here, it wouldn't make much sense to split those (I missed the dependency earlier) unless those are set up as useCallbacks()